src/server/trace/recorder/tracing.ts

```diff
@@ -194,42 +238,58 @@ export class Tracing implements InstrumentationListener {
             height: params.height,
             timestamp: monotonicTime()
           };
-          this._appendTraceEvent(event);
-          this._appendEventChain = this._appendEventChain.then(async () => {
+          // Make sure to write the screencast frame before adding a reference to it.
+          this._appendTraceOperation(async () => {
             await fs.promises.writeFile(path.join(this._resourcesDir!, sha1), params.buffer).catch(() => {});
           });
```

Comment:
We should probably catch here to avoid unhandled promise rejections.

Reply:
We do in line 243.